### PR TITLE
Alerting: Add Ref ID to DatasourceNoData and DatasourceError alerts

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What this PR does / why we need it**:

This pull request adds the Ref ID to `DatasourceNoData` and `DatasourceError` alerts.

Screenshot 2021-12-02 at 14 25 14
Screenshot 2021-12-02 at 14 26 07

### CODE REVIEW

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

File: pkg/services/ngalert/eval/eval.go [hunk 1]
```diff
@@ -317,12 +317,11 @@ func evaluateExecutionResult(execResults ExecutionResults, ts time.Time) Results
...
-		m := make(map[string]struct{})
-		for _, datasourceUID := range execResults.NoData {
-			if _, ok := m[datasourceUID]; !ok {
-				appendNoData(data.Labels{"datasource_uid": datasourceUID})
-				m[datasourceUID] = struct{}{}
-			}
+		for refID, datasourceUID := range execResults.NoData {
+			appendNoData(data.Labels{
+				"datasource_uid": datasourceUID,
+				"ref_id":         refID,
```
Comment (gotjosh) [inline pkg/services/ngalert/eval/eval.go:323]: nit: how do you feel about these two being constants?

**Review by gotjosh — APPROVED** [review]
LGTM

**Comment by yuri-tceretian** [comment]
For example, an alert rule that queries the same data source many times (eg ref_id A, B and C). NoData can produce multiple alerts per query. I think we should group ref_id by data source UID, ie the label would look like `ref_id=A,B,C`.

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Pull request merged**